great_expectations/dataset/base.py

```diff
@@ -821,10 +839,85 @@ def validate(self, expectations_config=None, catch_exceptions=True, result_forma
                     abbrev_results.append(exp)
             results = abbrev_results
 
-        return {
-            "results" : results
-        }
+        if evaluation_parameters is not None:
+            return {
+                "results": results,
+                "evaluation_parameters": evaluation_parameters
+            }
+        else:
+            return {
+                "results": results
+            }
+
+    def get_evaluation_parameter(self, parameter_name, default_value=None):
+        """Get an evaluation parameter value that has been stored in meta.
+        
+        Args:
+            parameter_name (string): The name of the parameter to store.
+            default_value (any): The default value to be returned if the parameter is not found.
+
+        Returns:
+            The current value of the evaluation parameter.
+        """
+        if "meta" in self._expectations_config and \
+                "evaluation_parameters" in self._expectations_config["meta"] and \
+                parameter_name in self._expectations_config['meta']['evaluation_parameters']:
+            return self._expectations_config['meta']['evaluation_parameters'][parameter_name]
+        else:
+            return default_value
+
+    def set_evaluation_parameter(self, parameter_name, parameter_value):
+        """Provide a kwarg value to be stored in the dataset meta object and used to evaluate expectations with missing
+        parameters.
+
+        Args:
+            expectation_name (string): The name of the expectation for which this kwarg should be used
+            kwarg_name (string): The name of the kwarg to be replaced at evaluation time
+            kwarg_value (any): The value to be used
+        """
+        # if not callable(getattr(self, expectation_name, None)):
```

Comment:
Looks like there's some cruft left over here...

Reply:
Yeah, I'll pull it out.

Reply:
Done.